node/network/statement-distribution/src/lib.rs

```diff
@@ -508,7 +508,10 @@ enum LargeStatementStatus {
 struct FetchingInfo {
 	/// All peers that send us a `LargeStatement` or a `Valid` statement for the given
 	/// `CandidateHash`, together with their originally sent messages.
-	available_peers: HashMap<PeerId, Vec<protocol_v1::StatementDistributionMessage>>,
+	///
+	/// We use an `IndexMap` here to preserve the ordering of peers sending us messages. This is
+	/// desirable because we reward first sending peers with reputation.
```

Comment:
Where is this needed?

Reply:
[here](https://github.com/paritytech/polkadot/blob/eae1054f06d0489f2d4dc33f04f263292ca8694d/node/network/statement-distribution/src/lib.rs#L1369)

Reply:
Doesn't work

Reply:
Sorry, link just became live with the push.

Reply:
Ahh, now I see how this works :P 

I had found the position before where this reputation change was done, but I didn't followed the full call chain to see how this is connected to `available_peers`.

Reply:
Sorry had to rush that req/resp PR quite a bit, so I am afraid I did not exactly make it more readable. :-S